pkg/apis/core/validation/seed.go

```diff
@@ -284,46 +284,90 @@ func validateSeedNetworks(seedNetworks core.SeedNetworks, fldPath *field.Path, i
 		return append(allErrs, errs...)
 	}
 
+	// Four buckets: all cidrs, seed cidrs and shoot defaults V4 and V6
+	// - all cidrs: must be valid CIDRs, must not overlap with reserved ranges
+	// - seed cidrs: must not overlap with each other
+	// - shoot defaults (V4): must not overlap with each other
+	// - shoot defaults (V6): must not overlap with each other, must not overlap with seed networks
 	var (
-		primaryIPFamily          = helper.DeterminePrimaryIPFamily(seedNetworks.IPFamilies)
-		networks                 []cidrvalidation.CIDR
-		reservedSeedServiceRange = cidrvalidation.NewCIDR(v1beta1constants.ReservedKubeApiServerMappingRange, field.NewPath(""))
+		primaryIPFamily                        = helper.DeterminePrimaryIPFamily(seedNetworks.IPFamilies)
+		allCIDRs                               []cidrvalidation.CIDR
+		seedCIDRs                              []cidrvalidation.CIDR
+		shootDefaultsV4                        []cidrvalidation.CIDR
+		shootDefaultsV6                        []cidrvalidation.CIDR
+		vpnRangeV6                             = cidrvalidation.NewCIDR(v1beta1constants.DefaultVPNRangeV6, field.NewPath(""))
+		reservedKubeApiServerMappingRange      = cidrvalidation.NewCIDR(v1beta1constants.ReservedKubeApiServerMappingRange, field.NewPath(""))
+		reservedSeedPodNetworkMappedRange      = cidrvalidation.NewCIDR(v1beta1constants.ReservedSeedPodNetworkMappedRange, field.NewPath(""))
```

Comment:
It seems like some of these validations are already implemented in https://github.com/gardener/gardener/blob/bd26073d53dd985aeb88bdd0ec03a5f87e7d06f8/pkg/utils/validation/cidr/disjoint.go (used by gardener-scheduler).
Is there a way to reuse the validation code to prevent mismatches in multiple implementations?

Reply:
the code in disjoint.go is very specific for shoots, not shootDefaults in Seeds. 
but I see your point, I'll try to refactor so that it all fits together.

Reply:
@timebertt I think it should work now. I've generalised the reserved ranges check so that it works for both seeds and shoots now. So the code for that is now in one place only.